### PR TITLE
feature: Masterpoint component tasks

### PR DESCRIPTION
**Pull request by gberenice**
## Info
This adds 2 tasks to manage Masterpoint [terraform-components ](https://github.com/masterpointio/terraform-components)
1. `init`
  * Fetches the component's configuration template from our central repository.
  * Replaces placeholders with the actual component name and version.
  * Sets up a directory structure to manage the component within the local environment by calling `update` task.
2. `update` 
  * Fetch the component's sources from our central repository bases on `component.yaml` data.
  * Cleans up all the temporary resources.

Also, this upgrades trunk and adds a fix for our lint pipeline.

### CODE REVIEW

**Comment by gberenice** [comment]
Hm, @kevcube said [he fixed this issue](https://masterpoint.slack.com/archives/C035S6TE7PC/p1697551741320799?thread_ts=1697462944.163189&cid=C035S6TE7PC) by upgrading `trunk` version in lint, but it fails in this repo:
```sh
✖ Failed to post annotations to github: GitHub Error:401 Body:{"message":"Bad credentials","documentation_url":"https://docs.github.com/rest"}
Error: Process completed with exit code 1.
```

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -1,12 +1,17 @@
 name: Lint
 
 on: pull_request
-permissions: read-all
+
+permissions:
+  actions: read
+  checks: write
```
Comment (Gowiem) [inline .github/workflows/lint.yaml:7]: Just so I'm aware -- what do we need `write` on checks for?

File: lib/components/Taskfile.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  update:
+    desc: Update an existing component from its source based on the config information.
+    requires:
+      vars:
+        - COMPONENT_NAME
+    vars:
+      COMPONENT_PATH: "components/{{.COMPONENT_NAME}}"
+      COMPONENT_CONFIG: "{{.COMPONENT_PATH}}/component.yaml"
+      COMPONENT_SOURCE_URI:
+        sh: yq eval '.component.source.uri' {{.COMPONENT_CONFIG}}
```
Comment (Gowiem) [inline lib/components/Taskfile.yml:30]: We should start moving towards [`dasel`](https://daseldocs.tomwright.me/examples/basics). Not necessary as part of this PR, but just calling it out.

**Review by Gowiem — COMMENTED** [review]
Couple questions!

File: lib/components/Taskfile.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      - mkdir -p ./components/{{.COMPONENT_NAME}}
+      - echo 'Fetching component config file...'
+      - |
+        curl -L https://raw.githubusercontent.com/masterpointio/terraform-components/$COMPONENT_VERSION/templates/component.yaml \
+        | sed -e 's|COMPONENT_NAME|{{.COMPONENT_NAME}}|g' -e 's|COMPONENT_VERSION|{{.COMPONENT_VERSION}}|g' \
+        > ./components/{{.COMPONENT_NAME}}/component.yaml
+      - task: update
+
+  update:
+    desc: Update an existing component from its source based on the config information.
```
[the remark this answers is not in the document]
Comment (gberenice) [reply lib/components/Taskfile.yml:22]: I assume that the `component.yaml` is updated with `init` task or manually.
`update` task just uses it as a source for commands to run - we're pulling the components sources according to the uri, name and version specified in the `component.yaml`. 
The sequence is:
- it pulls the repo in `tarball`
- extracts it
- moves the component sources to the component path in working directory
- removes the files we don't need anymore.

Let's discuss root config in our Internal call, I'll check out the atmos updates.

File: .github/workflows/lint.yaml (hunk 1 above)
Comment (gberenice) [reply .github/workflows/lint.yaml]: It was initially mentioned in [this thread](https://masterpoint.slack.com/archives/C035S6TE7PC/p1697462944163189) that this action need write permissions to post annotations.
I've just reproduced in for this repo https://github.com/masterpointio/taskit/actions/runs/6669078751/job/18126063384#step:3:338

**Review by kevcube — APPROVED** [review]

**Review by Gowiem — APPROVED** [review]

**Pull request merged**